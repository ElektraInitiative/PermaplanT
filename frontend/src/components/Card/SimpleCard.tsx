interface SimpleCardProps {
  /** title of the card */
  title: string;
  /** text paragraph */
  body: string;
}

/**
 * Simple text card component
 * @param props.title title of the card
 * @param props.body text paragraph
 */
export default function SimpleCard(props: SimpleCardProps) {
  return (
    <div>
      <h2 className="mb-2 text-sm font-medium">{props.title}</h2>
      <p>{props.body}</p>
    </div>
  );
}
