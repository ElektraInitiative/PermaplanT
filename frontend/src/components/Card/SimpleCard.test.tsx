import SimpleCard from './SimpleCard';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <SimpleCard title="Permaplant" body="This is a test." />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
