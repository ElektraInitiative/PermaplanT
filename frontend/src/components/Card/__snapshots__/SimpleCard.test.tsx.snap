// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div>
  <h2
    className="mb-2 text-sm font-medium"
  >
    Permaplant
  </h2>
  <p>
    This is a test.
  </p>
</div>
`;
